miso-web/src/main/webapp/scripts/shared_hot.js

```diff
@@ -0,0 +1,264 @@
+/**
+* Module for Handsontsble code which is shared between multiple instances
+*/
+var Hot = {
+  detailedSample: null,
+  colConf: null,
+  sampleOptions: null,
+  hotTable: null,
+  startData: [],
+  dropdownRef: null,
+  messages: {
+    success: [],
+    failed: []
+  },
+  failedComplexValidation: [],
+  saveButton: null,
+
+  /**
+   * Gets data needed to fill in dropdowns for various sample/library attributes. Callback is generally a "make table"-type function call.
+   */
+  fetchSampleOptions: function (callback) {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function() {
+      if (xhr.readyState === XMLHttpRequest.DONE) {
+        if (xhr.status === 200) {
+          Hot.sampleOptions = JSON.parse(xhr.responseText);
+          callback();
+        } else {
+          console.log(xhr.response);
+        }
+      }
+    };
+    xhr.open('GET', '/miso/rest/ui/sampleoptions');
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    xhr.send();
+  },
+
+  /**
+   * Gets item's alias
+   */
+  getAlias: function (obj) {
+    if (obj.alias) return obj.alias;
+  },
+
+  /**
+   * Gets values for a given key
+   */
+  getValues: function (key, objArr) {
+    return objArr.map(function (obj) { return obj[key]; });
+  },
+
+  /**
+   * Sorts by a given property
+   */
+  sortByProperty: function (array, propertyName) {
+    return array.sort(function (a, b) {
+      return a[propertyName] > b[propertyName] ? 1 : ((b[propertyName] > a[propertyName]) ? -1 : 0);
+    });
+  },
+
+  /**
+   * Gets the id of an object from a given alias and collection
+   */
+  getIdFromAlias: function (alias, referenceCollection) {
+    return referenceCollection.filter(function (item) {
+      return item.alias == alias;
+    })[0].id;
+  },
+
+  /**
+   * Gets the alias of an object from a given id and collection
+   */
+  getAliasFromId: function (id, referenceCollection) {
+    return referenceCollection.filter(function (item) {
+      return item.id == id;
+    })[0].alias;
```

Comment:
This might do well with a check for length to avoid the case where the id isn't present.